### PR TITLE
Testing branch

### PR DESCRIPTION
**Pull request by egertje**
Here I made the rest of the tests for DataManager and TransactionEngine

### CODE REVIEW

File: src/transactions/TransactionEngineTest.java [hunk 1]
```diff
@@ -43,6 +46,7 @@ public void testUniquenessAndDigits() {
             //transaction IDs are all 24 digits, are alpha numeric, and are all unique
         String[] transactionIDs = new String[1000];
         for(int i = 0; i < transactionIDs.length; i++) {
+            //we recreated this Person object inside the for loop so a new transaction ID would be generated each time
```
Comment (osilkin98) [inline src/transactions/TransactionEngineTest.java:49]: I like this style of commenting

File: src/transactions/TransactionEngineTest.java [hunk 2]
```diff
@@ -55,5 +59,50 @@ public void testUniquenessAndDigits() {
...
+    @Test
+    public void testIDCreatedWithGetPersonMap() {
+        DataManager dataManager = new DataManager("Test-Customers.csv");
+        //now do the createPersonMap() method to create a map of Person objects with their account number
+            //as their key
+        Map<String, Person> personMap = dataManager.createPersonMap();
+        //Now I will get a few of the Person objects in the csv file and store them here to see if there are
+            //transaction IDs for them
+        Person person1 = personMap.get("abc123");
+        Person person2 = personMap.get("success789");
```
Comment (osilkin98) [inline src/transactions/TransactionEngineTest.java]: In general, you want to write this like
```py 
# generate & test person 1 in this block
person1 = generatePerson()
assertSomething(person2)

# test stuff about person 2 over here
person2 = generateAnotherPerson()
assertSomethingElse(person2)
```
This way it’s easier to track what breaks.
Besides that, this looks good.

File: src/transactions/TransactionEngineTest.java [hunk 3]
```diff
@@ -55,5 +59,50 @@ public void testUniquenessAndDigits() {
...
+        assertNotNull(person3.getTransactionId());
+    }
+    //Now I will test the getAllCustomerTransactionIDs and getCustomerTransactionID methods
+    //To do this I created a csv file with a few Person objects stored in it and am using this csv file to create
+        //a DataManager object to test these methods
+    @Test
+    public void testGetAllCustomerTransactionIDs() {
+        TransactionEngine transactions = new TransactionEngine("Test-Customers.csv");
+        //to test getAllCustomerTransactionIDs I will make sure the output of this method isn't null
+        assertNotNull(transactions.getAllCustomerTransactionIDs());
```
Comment (osilkin98) [inline src/transactions/TransactionEngineTest.java:93]: Good, though I would make sure that the contents of the method are also not null / are proper transaction ids

File: src/transactions/TransactionEngineTest.java [hunk 4]
```diff
@@ -55,5 +59,50 @@ public void testUniquenessAndDigits() {
...
+        assertNotNull(person2.getTransactionId());
+        //person3 has no empty fields in the CSV file
+        assertNotNull(person3.getTransactionId());
+    }
+    //Now I will test the getAllCustomerTransactionIDs and getCustomerTransactionID methods
+    //To do this I created a csv file with a few Person objects stored in it and am using this csv file to create
+        //a DataManager object to test these methods
+    @Test
+    public void testGetAllCustomerTransactionIDs() {
+        TransactionEngine transactions = new TransactionEngine("Test-Customers.csv");
```
Comment (osilkin98) [inline src/transactions/TransactionEngineTest.java:91]: I would recommend hard-coding the CSV file somewhere in the same test, and then writing it to the file system & passing its path here. Also try passing in the CSV in its string format directly to the data manager somehow. If we can’t do this, then we should rethink our design for the `DataManager` class

**Review by osilkin98 — APPROVED** [review]
Good work all around. I left a few comments, but I really like what I see.

File: src/transactions/TransactionEngineTest.java (hunk 2 above)
Comment (Meantub) [reply src/transactions/TransactionEngineTest.java]: Yeah this will make it easier to debug should something go wrong

Comment (egertje) [reply src/transactions/TransactionEngineTest.java]: Okay, I can change that stuff tomorrow

File: src/transactions/TransactionEngineTest.java (hunk 3 above)
Comment (egertje) [reply src/transactions/TransactionEngineTest.java:93]: What do you mean by that Oleg?

File: src/transactions/TransactionEngineTest.java (hunk 4 above)
Comment (egertje) [reply src/transactions/TransactionEngineTest.java:91]: Okay, I can try that tomorrow

File: src/transactions/TransactionEngineTest.java (hunk 3 above)
Comment (osilkin98) [reply src/transactions/TransactionEngineTest.java:93]: You assert that the list being returned isn't `null`, but you haven't asserted that each of its elements are not `null`.

File: src/transactions/TransactionEngineTest.java (hunk 4 above)
Comment (osilkin98) [reply src/transactions/TransactionEngineTest.java:91]: Actually, don't worry about it

**Review by CMontani89 — COMMENTED** [review]
looks good'

**Pull request merged**